agents/docs/hardware-autoresearch.md

```
@@ -210,6 +210,25 @@ reflexively can hurt:
 - **ESP32 / Teensy / LPC** benches usually *do* want an explicit port, because
   several identical boards are typically attached.
 
+An explicit `--upload-port` is checked for presence *before* the build, because
+a port that is not attached can only fail — and it used to fail at deploy, after
+roughly seven minutes of package install, lint and build:
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)